src/plugins/discourse/mirror.js

```diff
@@ -67,15 +73,109 @@ export class Mirror {
       ...defaultOptions,
       ...(options || {}),
     };
+    this._mode = mode || 1;
   }
 
   async update(reporter: TaskReporter) {
     reporter.start("discourse");
-    await this._updateTopics(reporter);
-    await this._updateLikes(reporter);
+    switch (this._mode) {
+      case 1:
+        await this._updateTopics(reporter);
+        await this._updateLikes(reporter);
+        break;
+      case 2:
+        await this._updateTopicsV2(reporter);
+        await this._updateLikes(reporter);
+        break;
+      default:
+        throw new Error(`Invalid mode ${this._mode}, expecting 1 or 2`);
+    }
     reporter.finish("discourse");
   }
 
+  async _updateTopicsV2(reporter: TaskReporter) {
+    reporter.start("discourse/topics");
+
+    const {
+      topicBumpMs: lastLocalTopicBumpMs,
+      definitionCheckMs: lastDefinitionCheckMs,
+    } = this._repo.syncHeads();
+
+    const startTime = Date.now();
+    const shouldCheckDefinitions =
+      startTime - lastDefinitionCheckMs >=
+      this._options.recheckCategoryDefinitionsAfterMs;
+
+    const bumpedTopics: TopicLatest[] = await this._fetcher.topicsBumpedSince(
+      lastLocalTopicBumpMs
+    );
+    const topicBumpsById: Map<number, number> = new Map(
+      bumpedTopics.map((t) => [t.id, t.bumpedMs])
+    );
+
+    // Make sure we have oldest first in our load queue.
+    bumpedTopics.sort(sortTopicByBumpedMs);
```

Comment:
Hm, reason I tend to avoid this is because it's a [2630 line dependency](https://github.com/lodash/lodash/blob/4.7.0-npm-packages/lodash.sortby/index.js) to abstract away a 3 line comparison function and shallow copy instead of mutate. Although I'll agree that comparison functions require a bit more working memory to reason about, I think that's much alleviated by not inlining the comparison function.

Reply:
For `lodash.sortby` I'd like to have a go at replacing it with a small unit function. But since this is definitely not the only place where it would need to be used, I think that is best done as a separate PR. For now I feel this solution is way more comprehensible than the dependency.